src/platform/graphics/constants.js

```diff
@@ -1179,6 +1179,18 @@ export const DEVICETYPE_WEBGL2 = 'webgl2';
  */
 export const DEVICETYPE_WEBGPU = 'webgpu';
 
+/**
+ * Enum for the device types.
+ *
+ * @readonly
+ * @enum {string}
+ */
+export const DEVICETYPE = {
+    WEBGL1: DEVICETYPE_WEBGL1,
+    WEBGL2: DEVICETYPE_WEBGL2,
+    WEBGPU: DEVICETYPE_WEBGPU
+};
```

Comment:
I'm a little confused why we have now the API `DEVICETYPE_XXX` and `DEVICETYPE.XXX` (the latter being out of step with the rest of the API constants). @mvaligursky, this request for an enum was from you, right? Can you clarify what you wanted a bit further?

Reply:
my hope is to have a single type, or a similar solution, to avoid this in every example:
'''
deviceType: typeof pc.DEVICETYPE_WEBGPU | typeof pc.DEVICETYPE_WEBGL1 | typeof pc.DEVICETYPE_WEBGL2
'''

Reply:
It seems like we can either list out all the different options when typing the deviceType property, be less specific with the type of that property or use an enum to group the different device types together.

The enum works pretty well in this instance but like @willeastcott said it'd be the first instance of `@enum` in the codebase.

Reply:
By enum I assumed a type created in the example browser, not in the engine, as that type is only even used by the example browser.

Reply:
Yeah, that's what I thought you meant too. In the engine source, I have no idea what `@enum` would generate in terms of JSDoc output (API reference) and TS declarations.

Reply:
I don't think it should, as current implementation matches the rest of this engine, while this makes it different.
If at some point we decide to adapt enums in some form, it would probably need to be an engine-wide decision.

Reply:
Ok, i've set the type of the `deviceType` property to string for now.